api/apps/v1alpha1/nemo_datastore_types.go

```diff
@@ -157,97 +169,319 @@ func (n *NemoDatastore) GetStandardLabels() map[string]string {
-	return []corev1.EnvFromSource{
+// GetVolumes returns volumes for the NemoDatastore container
+func (n *NemoDatastore) GetVolumes() []corev1.Volume {
+	/*volumes:
+	  - name: init
+	    secret:
+	      defaultMode: 110
+	      secretName: datastore-nemo-datastore-init
+	  - name: config
+	    secret:
+	      defaultMode: 110
+	      secretName: datastore-nemo-datastore
+	  - name: inline-config-sources
+	    secret:
+	      defaultMode: 420
+	      secretName: datastore-nemo-datastore-inline-config
+	  - emptyDir: {}
+	    name: temp
+	  - name: data
+	    persistentVolumeClaim:
+	      claimName: datastore-shared-storage
+	*/
+	var initMode = int32(110)
+	var configMode = int32(420)
+
+	volumes := []corev1.Volume{
 		{
-			ConfigMapRef: &corev1.ConfigMapEnvSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: n.Spec.DataStoreParams.EnvConfigMap,
+			Name: "init",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  n.Spec.DataStoreParams.DataStoreInitSecret,
+					DefaultMode: &initMode,
 				},
 			},
 		},
 		{
-			SecretRef: &corev1.SecretEnvSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: n.Spec.DataStoreParams.EnvSecret,
+			Name: "config",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  n.Spec.DataStoreParams.DataStoreConfigSecret,
+					DefaultMode: &initMode,
+				},
+			},
+		},
+		{
+			Name: "inline-config-sources",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  n.Spec.DataStoreParams.DataStoreInlineConfigSecret,
+					DefaultMode: &configMode,
 				},
 			},
 		},
+		{
+			Name: "temp",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
 	}
+
+	if n.ShouldEnablePersistentStorage() {
+		volumes = append(volumes, corev1.Volume{
+			Name: "data",
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: n.Spec.DataStoreParams.Persistence.ClaimName,
+				},
+			},
+		})
+	} else {
+		volumes = append(volumes, corev1.Volume{
+			Name: "data",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		})
+	}
+	return volumes
 }
 
-func (n *NemoDatastore) GetInitContainers() []corev1.Container {
-	image := n.Spec.DataStoreParams.InitContainerImage
-	if image == "" {
-		image = "busybox"
-	}
-	cmd := n.Spec.DataStoreParams.InitContainerCommand
-	if len(cmd) == 0 {
-		cmd = []string{
-			"sh",
-			"-c",
-			fmt.Sprintf("until nc -z %s %s; do echo \"PostgreSQL is unavailable. Sleeping for 5 seconds\"; sleep 5; done;", n.Spec.DataStoreParams.DatabaseHost, n.Spec.DataStoreParams.DatabasePort),
-		}
+// GetPVC returns the persistence PVC object for the NeMoDatastore container
+func (n *NemoDatastore) GetPVC() (*corev1.PersistentVolumeClaim, error) {
```

Comment:
we can instead use shared.ConstructPVC() method directly in the controller package: https://github.com/NVIDIA/k8s-nim-operator/blob/main/internal/shared/pvc.go#L29